src/goalServices/goals.js

```diff
@@ -1527,17 +1527,31 @@ export async function goalsForGrants(grantIds) {
       'onApprovedAR',
       'endDate',
       'source',
+      'createdVia',
       [sequelize.fn('BOOL_OR', sequelize.literal(`"goalTemplate"."creationMethod" = '${CREATION_METHOD.CURATED}'`)), 'isCurated'],
     ],
-    group: ['"Goal"."name"', '"Goal"."status"', '"Goal"."endDate"', '"Goal"."onApprovedAR"', '"Goal"."source"'],
+    group: ['"Goal"."name"', '"Goal"."status"', '"Goal"."endDate"', '"Goal"."onApprovedAR"', '"Goal"."source"', '"Goal"."createdVia"'],
     where: {
+      name: {
+        [Op.ne]: '', // exclude "blank" goals
```

Comment:
This is just something that bothers me about our current selector - there always seems to be "blank" goals in there. I can remove from this PR if it's controversial as it isn't directly related to fixing the issue I found

Reply:
Makes me think we should be preventing saving of blank goals....

Reply:
We need to in order to save metadata underthem

Reply:
@GarrettEHill you are answering Adam's comment right? Or are you objecting to the code I added here?